src/main.ts

```diff
@@ -1,19 +1,39 @@
-import {getInput, debug, addPath, setFailed} from "@actions/core";
-import {downloadTool, extractZip} from "@actions/tool-cache";
-import {GitHub} from "@actions/github";
-import {resolve} from "path";
-import {exec} from "@actions/exec";
+import { getInput, debug, addPath, setFailed } from "@actions/core";
+import { downloadTool, extractZip } from "@actions/tool-cache";
+import { GitHub } from "@actions/github";
+import { resolve } from "path";
+import { exec } from "@actions/exec";
+import configFile from "./configFile";
 import foreman from "./foreman";
 
 async function run(): Promise<void> {
   try {
     const versionReq: string = getInput("version");
     const githubToken: string = getInput("token");
     const workingDir: string = getInput("working-directory");
+    const allowExternalGithubOrgs: string = getInput(
+      "allow-external-github-orgs"
+    ).toLowerCase();
+
+    if (allowExternalGithubOrgs != "true") {
+      debug("Checking tools in Foreman Config come from source org");
+      if (!process.env.GITHUB_REPOSITORY) {
+        throw new Error(
+          `Could not find repository setup-foreman is running in`
+        );
+      }
+      const repository = process.env.GITHUB_REPOSITORY.split("/");
+      if (repository.length == 0) {
+        throw new Error(
+          `Could not find owner of repository setup-foreman is running in`
+        );
+      }
+      const org = repository[0];
+      configFile.checkSameOrgInConfig(org.toLowerCase());
```

Comment:
```suggestion
      debug("Checking tools in Foreman Config come from source org");
      const owner = process.env.GITHUB_REPOSITORY_OWNER;
      if (!owner) {
        throw new Error(
          `Could not find repository owner setup-foreman is running in`
        );
      }
      configFile.checkSameOrgInConfig(owner.toLowerCase());
```
There is `GITHUB_REPOSITORY_OWNER` according to https://docs.github.com/en/actions/learn-github-actions/variables